File: Help Desk Ticketing System UI/HelpDeskTicketingSystemUI/src/app/models/tickets.ts
import { TicketStatus, TicketPriority } from './enums';
import { User } from './users';


export interface Ticket {
  id: number;
  title: string;
  description: string;
  status: TicketStatus;
  priority: TicketPriority;
  createdBy: number; // User ID
  assignedTo?: number; // Agent ID (optional)
  createdAt: string;
  updatedAt: string;

  // Optional UI properties (filled after fetching related data)
  createdByUser?: User;
  assignedToUser?: User;
}

File: Help Desk Ticketing System Server/src/main/java/org/manh/help_desk_ticketing_system_server/tickets/entity/Tickets.java
package org.manh.help_desk_ticketing_system_server.tickets.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.manh.help_desk_ticketing_system_server.auth.entity.Users;

import java.time.LocalDateTime;

enum TicketStatus {
    Open, InProgress, Resolved, Closed
}

enum TicketPriority {
    Low, Medium, High
}

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "tickets")
public class Tickets {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private String title;
    private String description;

    @Enumerated(EnumType.STRING)
    private TicketStatus status;

    @Enumerated(EnumType.STRING)
    private TicketPriority priority;

    @ManyToOne
    @JoinColumn(name = "created_by", nullable = false)
    private Users createdBy;

    @ManyToOne
    @JoinColumn(name = "assigned_to", nullable = true)
    private Users assignedTo;

    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
